onvm/onvm_nflib/onvm_nflib.c

```diff
@@ -947,6 +947,10 @@ onvm_nflib_dequeue_messages(struct onvm_nf_local_ctx *nf_local_ctx) {
 
         // Check and see if this NF has any messages from the manager
         if (likely(rte_ring_count(msg_q) == 0)) {
+                if (ONVM_NF_SHARE_CORES) {
+                        rte_atomic16_set(nf_local_ctx->nf->shared_core.sleep_state, 1);
+                        sem_wait(nf_local_ctx->nf->shared_core.nf_mutex);
```

Comment:
I'm not sure If I like the fact that it goes to sleep here + when it receives no packets, we should have some sort of benchmarks to verify that. Might be a good quick thing to discuss at the meeting

Reply:
Yea..maybe we can let the user specify what they want to sleep on (i.e, default for packets)